src/transport/raw/TCP.cpp

```diff
@@ -324,11 +326,11 @@ CHIP_ERROR TCPBase::ProcessReceivedBuffer(Inet::TCPEndPoint * endPoint, const Pe
         {
             return err;
         }
-        uint16_t messageSize = LittleEndian::Get16(messageSizeBuf);
-        if (messageSize >= kMaxMessageSize)
+        uint32_t messageSize = LittleEndian::Get32(messageSizeBuf);
+        if (messageSize >= kMaxTCPMessageSize)
         {
-
-            // This message is too long for upper layers.
+            // Message is too big for node to process. Disconnect with peer.
+            CloseConnectionInternal(state, CHIP_ERROR_MESSAGE_TOO_LONG, SuppressCallback::No);
```

Comment:
Again: this needs to be in a separate PR.  One PR for the mechanical "allow larger sizes" changes that affect all codepaths.  A separate PR for a significant change to the TCP behavior.  Please don't roll fundamentally unrelated changes all into the same PR.

Reply:
Removed the connection closure. Will float a separate PR after this is merged.

Reply:
Removed from this PR.Will float a separate one after this is merged.